### PR TITLE
Enable Text Analytics cilent to authenticate with token credential in USGOV and China

### PR DESCRIPTION
**Pull request by zedy-wj**
1.Fix issue https://github.com/Azure/azure-sdk-for-net/issues/18520
2.The scope is hard code leading to only work well in the public cloud.
image
Remove hardcoded value and enable Form Recognizer tests in UsGov and China clouds.

The value of the scope in different clouds is follow:
Azure public cloud: https://cognitiveservices.azure.com/.default
USGOV cloud: https://cognitiveservices.azure.us/.default
China cloud: https://cognitiveservices.azure.cn/.default

@jongio , @maririos for notification.

### CODE REVIEW

**Comment by jongio** [comment]
@zedy-wj @v-xuto - Can you research the CI failures?

**Comment by zedy-wj** [comment]
@jongio - Change service version not stable at  beginning, it passed in the last time.

**Comment by maririos** [comment]
Same comments of https://github.com/Azure/azure-sdk-for-net/pull/20970 apply here, so just to not repeat, I will put them in the FR PR

**Review by maririos — APPROVED** [review]

**Comment by maririos** [comment]
We are in code freeze until tomorrow. Once it is safe, I can go ahead and merge this PR.
Thanks @zedy-wj for all the work :)

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs [hunk 1]
```diff
@@ -22,7 +22,7 @@ public class TextAnalyticsClient
         internal readonly TextAnalyticsRestClient _serviceRestClient;
         internal readonly ClientDiagnostics _clientDiagnostics;
         private readonly TextAnalyticsClientOptions _options;
-        private readonly string DefaultCognitiveScope = "https://cognitiveservices.azure.com/.default";
+        private readonly string DefaultCognitiveScope = TextAnalyticsClientOptions.GetDefaultCognitiveScope(Environment.GetEnvironmentVariable("AZURE_AUTHORITY_HOST"));
```
Comment (schaabs) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs:25]: This won't be a reliable way of determining the cloud. Users can set the `AuthorityHost` via code as well as via the environment variable. If the scope can't be determined via the resource URI, and the service doesn't have a challenge we might need to add something like `AuthenticationScope` to the client options.

Comment (zedy-wj) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs]: I have updated codes, users can set different clouds at [here](https://github.com/zedy-wj/azure-sdk-for-net/blob/TA-sov-enable/sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClientOptions.cs#L20). Could you review it please? I am not sure whether this is the recommended method.

**Comment by jongio** [comment]
Closing this for now as we discuss design options.

**Pull request closed without merging**